### PR TITLE
Attempt to determine the correct image mode from the pixel format

### PR DESCRIPTION
**Pull request by agfor**
VMWare's built-in VNC server seems to ignore `SetPixelFormat` messages.
However, it does correctly indicate the pixel format it uses in the
`ServerInit` message.

This commit attempts to use the pixel format info from that initial
message to derive the correct image mode to use for PIL. If that fails,
it falls back to the previous behavior of setting the pixel format to
the `rfb.py` default of `RGBX`.

This was the least-hacky way I could think of to fix VMWare screenshots,
while potentially giving some benefit to other implementations (skipping
the often unnecessary, if harmless, `SetPixelFormat` message).

### CODE REVIEW

File: vncdotool/client.py [hunk 1]
```diff
@@ -336,6 +337,22 @@ def mouseDrag(self, x, y, step=1):
 
         return self
 
+    def setImageMode(self):
+        """ Extracts color ordering and 24 vs. 32 bpp info out of the pixel format information
+        """
+        if (self.truecolor and (not self.bigendian) and self.depth == 24
+                and self.redmax == 255 and self.greenmax == 255 and self.bluemax == 255):
```
Comment (sibson) [inline vncdotool/client.py]: To check I'm understanding this correctly, this is testing we're using a 3 channel 8bit colour map?

File: vncdotool/client.py [hunk 2]
```diff
@@ -347,7 +364,9 @@ def vncRequestPassword(self):
         self.sendPassword(self.factory.password)
 
     def vncConnectionMade(self):
-        self.setPixelFormat()
+        if not self.setImageMode():
```
Comment (sibson) [inline vncdotool/client.py]: I'm finding this hard to parse. Would it make sense push setPixelFormat() inside setImageMode() to avoid the negative conditional?

File: vncdotool/client.py (hunk 1 above)
Comment (agfor) [reply vncdotool/client.py]: Yeah. Basically, it can handle different color orderings, and with or without a padding / alpha byte, but that's it. So any attribute of the pixel format that I don't explicitly use to determine the image mode, I explicitly assert on here.

File: vncdotool/client.py (hunk 2 above)
Comment (agfor) [reply vncdotool/client.py]: Yeah, that's better. Updated.

File: vncdotool/client.py (hunk 1 above)
Comment (sibson) [reply vncdotool/client.py]: cool, I like it also opens the door to other formats being implemented, as needed. There are failing tests, seems like truecolor isn't being handled consistently.  Once those are fixed I'll merge this.
